CMakeLists.txt

```diff
@@ -36,7 +36,9 @@ if(ROS_VERSION EQUAL 1) # if ROS1
   generate_messages(DEPENDENCIES ${MESSAGE_DEPENDENCIES})
 
   add_library(${PROJECT_NAME} INTERFACE)
-
+  set_target_properties(${PROJECT_NAME} PROPERTIES
+      INTERFACE_INCLUDE_DIRECTORIES ${${PROJECT_NAME}_INCLUDE_DIRS}
+  )
```

Comment:
```suggestion
  target_include_directories(${PROJECT_NAME} PUBLIC $<INSTALL_INTERFACE:include>)
```

Reply:
I tested your suggestions and I got the following error:
```
CMake Error at CMakeLists.txt:39 (target_include_directories):
  target_include_directories may only set INTERFACE properties on INTERFACE
  targets
```
Works fine if I change it to:
```target_include_directories(${PROJECT_NAME} INTERFACE $<INSTALL_INTERFACE:include>)``` is that correct?